src/utils/json-big.ts

```diff
@@ -0,0 +1,62 @@
+import JsonBig from '@aeternity/json-bigint'
+import BigNumber from 'bignumber.js'
+
+const jsonBig = JsonBig({ storeAsString: true })
+
+interface BigNumValueObject {
+  jsonrpc: string
+  method: string
+  params?: {
+    action?: string
+    signed_tx?: string
+    code?: string
+    call_data?: string
+    deposit?: BigNumber
+    vm_version?: number
+    abi_version?: number
+    amount?: number
+    gas_price?: number
+    gas?: number
+    error?: number
+    caller_id?: string
+    contract_id?: string
+    round?: number
+    pubkey?: string
+    data?: {
+      channel_id?: string
+      error_code?: number
+      error_msg?: string
+      round?: number
+    }
+  }
+  id?: string
+  version?: number
+}
+
+type ConvertedValueType = BigNumValueObject | BigNumber | string
+
+const convertValuesToBigNumbers = (
+  value: ConvertedValueType): ConvertedValueType | {} => {
+  if (typeof value === 'object' && value !== null && value.constructor === Object) {
+    return Object.entries(value)
+      .map(([key, value]) => [key, convertValuesToBigNumbers(value)])
+      .reduce((p, [k, v]: [k:string, v:ConvertedValueType]) => ({ ...p, [k]: v }), {})
+  }
+  if (Array.isArray(value)) {
+    return value.map(item => convertValuesToBigNumbers(item))
+  }
+  if (typeof value === 'string' && new BigNumber(value).toString(10) === value) {
+    const bn = new BigNumber(value)
+    bn.toJSON = () => bn.toString(10)
+    return bn
+  }
+  return value
+}
+
+export default {
+  stringify: (
+    object: ConvertedValueType,
+    ...args: [replacer: Function | string[], space: string | number]
```

Comment:
I propose to refer to `jsonBig.stringify` type as far as we just proxying these arguments